src/neptune/objects/abstract.py

```diff
@@ -71,10 +71,10 @@ def get_root_object(self) -> "SupportsNamespaces":
         ...
 
 
-class NeptuneObject(SupportsNamespaces, ABC):
+class AbstractNeptuneObject(SupportsNamespaces, ABC):
```

Comment:
Probably this could be just replaced with `MetadataContainer` renamed to `NeptuneObject`. Check the usages.